src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java

```diff
@@ -320,4 +249,77 @@ public String getGoogleId() {
     public String getRole() {
         return role;
     }
+
+    /**
+     * A Builder class for {@link InstructorAttributes}.
+     */
+    public static class Builder {
+        private final InstructorAttributes instructorAttributes;
+
+        public Builder(String googleId, String courseId, String name, String email) {
+            instructorAttributes = new InstructorAttributes();
+
+            instructorAttributes.googleId = SanitizationHelper.sanitizeGoogleId(googleId);
+            instructorAttributes.courseId = SanitizationHelper.sanitizeTitle(courseId);
+            instructorAttributes.name = SanitizationHelper.sanitizeName(name);
+            instructorAttributes.email = email;
+
+            instructorAttributes.role = Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER;
+            instructorAttributes.displayedName = DEFAULT_DISPLAY_NAME;
+            instructorAttributes.isArchived = false;
+            instructorAttributes.isDisplayedToStudents = true;
+            instructorAttributes.privileges = new InstructorPrivileges(instructorAttributes.role);
+        }
+
+        public Builder withKey(String key) {
+            instructorAttributes.key = key;
+            return this;
+        }
+
+        public Builder withRole(String role) {
+            if (role != null) {
```

Comment:
I left ternary operator for privileges because the default value of that field is based on current value of role field. So every time when null passed to privileges method of the builder, will be used privilege based on current role.